sentry-android-core/src/test/java/io/sentry/android/core/SpanFrameMetricsCollectorTest.kt

```diff
@@ -139,42 +174,50 @@ class SpanFrameMetricsCollectorTest {
         val sut = fixture.getSut()
 
         // when the first span starts
-        val span0 = createFakeSpan()
+        fixture.timeNanos = 0
+        val span0 = createFakeSpan(0, 800)
         sut.onSpanStarted(span0)
 
         // and one fast, two slow frames and one frozen is are recorded
         sut.onFrameMetricCollected(0, 10, 10, 0, false, false, 60.0f)
-        sut.onFrameMetricCollected(0, 20, 20, 4, true, false, 60.0f)
-        sut.onFrameMetricCollected(0, 20, 20, 4, true, false, 60.0f)
-        sut.onFrameMetricCollected(0, 800, 800, 784, true, true, 60.0f)
+        sut.onFrameMetricCollected(16, 48, 32, 16, true, false, 60.0f)
+        sut.onFrameMetricCollected(60, 92, 32, 16, true, false, 60.0f)
+        sut.onFrameMetricCollected(100, 800, 800, 784, true, true, 60.0f)
 
         // then a second span starts
-        val span1 = createFakeSpan()
+        fixture.timeNanos = 800
+        sut.onSpanFinished(span0)
+
+        fixture.timeNanos = 820
+        val span1 = createFakeSpan(820, 840)
         sut.onSpanStarted(span1)
 
         // and another slow frame is recorded
-        sut.onFrameMetricCollected(0, 20, 20, 4, true, false, 60.0f)
-
-        sut.onSpanFinished(span0)
+        fixture.timeNanos = 840
+        sut.onFrameMetricCollected(820, 840, 20, 4, true, false, 60.0f)
         sut.onSpanFinished(span1)
```

Comment:
adding `sut.onSpanFinished(createFakeSpan(820, 840))` right before `sut.onSpanFinished(span1)` make it fail, due to concurrent spans not working
create another test specifically for it, after making it work